File: _includes/code/search.basics.js
const assert = require('assert');

// ================================
// ===== INSTANTIATION-COMMON =====
// ================================

// BasicGetJS
const { default: weaviate } = require('weaviate-ts-client');

// END BasicGetJS
// ===== Instantiation shown on snippet
// BasicGetJS
const client = weaviate.client({
  scheme: 'https',
  host: 'some-endpoint.weaviate.network',  // Replace with your Weaviate URL
  apiKey: new weaviate.ApiKey('YOUR-WEAVIATE-API-KEY'),  // If authentication is on. Replace w/ your Weaviate instance API key
});
// END BasicGetJS

// ==============================
// ===== BASIC GET EXAMPLES =====
// ==============================

// BasicGetJS
async function basicGetQuery() {
  try {
    const res = await client
      .graphql
      .get()
      .withClassName('JeopardyQuestion')
      .withFields('question')
      .do();
    console.log(JSON.stringify(res));
    return res;
  } catch (err) {
    console.error(err);
  }
}

basicGetQuery();
// END BasicGetJS

// Test
basicGetQuery().then(res => {
  assert('JeopardyQuestion' in res.data.Get);
  const questionKeys = new Set(Object.keys(res.data.Get.JeopardyQuestion[0]));
  assert.deepEqual(questionKeys, new Set(['question']));
});
// End test


// ===================================
// ===== GET WITH LIMIT EXAMPLES =====
// ===================================

// GetWithLimitJS
async function getWithLimitQuery() {
  try {
    const res = await client
      .graphql
      .get()
      .withClassName('JeopardyQuestion')
      .withFields('question')
      .withLimit(1)
      .do();
    console.log(JSON.stringify(res));
    return res;
  } catch (err) {
    console.error(err);
  }
}

getWithLimitQuery();
// END GetWithLimitJS

// Test
getWithLimitQuery().then(res => {
  assert('JeopardyQuestion' in res.data.Get);
  assert.deepEqual(res.data.Get.JeopardyQuestion.length, 1)
  const questionKeys = new Set(Object.keys(res.data.Get.JeopardyQuestion[0]));
  assert.deepEqual(questionKeys, new Set(['question']));
});
// End test


// ===================================
// ===== GET PROPERTIES EXAMPLES =====
// ===================================

// GetPropertiesJS
async function getPropertiesQuery() {
  try {
    const res = await client
      .graphql
      .get()
      .withClassName('JeopardyQuestion')
      .withFields('question answer points')
      .withLimit(1)
      .do();
    console.log(JSON.stringify(res));
    return res;
  } catch (err) {
    console.error(err);
  }
}

getPropertiesQuery();
// END GetPropertiesJS

// Test
getPropertiesQuery().then(res => {
  assert('JeopardyQuestion' in res.data.Get);
  assert.deepEqual(res.data.Get.JeopardyQuestion.length, 1)
  const questionKeys = new Set(Object.keys(res.data.Get.JeopardyQuestion[0]));
  assert.deepEqual(questionKeys, new Set(['question', 'answer', 'points']));
});
// End test


// ======================================
// ===== GET OBJECT VECTOR EXAMPLES =====
// ======================================

// GetObjectVectorJS
async function getObjectVectorQuery() {
  try {
    const res = await client
      .graphql
      .get()
      .withClassName('JeopardyQuestion')
      .withFields('_additional {vector}')
      .withLimit(1)
      .do();
    console.log(JSON.stringify(res));
    return res;
  } catch (err) {
    console.error(err);
  }
}

getObjectVectorQuery();
// END GetObjectVectorJS

// Test
getObjectVectorQuery().then(res => {
  assert('JeopardyQuestion' in res.data.Get);
  assert.deepEqual(res.data.Get.JeopardyQuestion.length, 1)
  const additionalKeys = new Set(Object.keys(res.data.Get.JeopardyQuestion[0]._additional));
  assert.deepEqual(additionalKeys, new Set(['vector']));
});
// End test


// ==================================
// ===== GET OBJECT ID EXAMPLES =====
// ==================================

// GetObjectIdJS
async function getObjectIdQuery() {
  try {
    const res = await client
      .graphql
      .get()
      .withClassName('JeopardyQuestion')
      .withFields('_additional {id}')
      .withLimit(1)
      .do();
    console.log(JSON.stringify(res));
    return res;
  } catch (err) {
    console.error(err);
  }
}

getObjectIdQuery();
// END GetObjectIdJS

// Test
getObjectIdQuery().then(res => {
  assert('JeopardyQuestion' in res.data.Get);
  assert.deepEqual(res.data.Get.JeopardyQuestion.length, 1)
  const additionalKeys = new Set(Object.keys(res.data.Get.JeopardyQuestion[0]._additional));
  assert.deepEqual(additionalKeys, new Set(['id']));
});
// End test

File: _includes/code/search.basics.py
# ================================
# ===== INSTANTIATION-COMMON =====
# ================================

# ===== Instantiation shown on snippet
import weaviate

client = weaviate.Client(
    "https://some-endpoint.weaviate.network",  # Replace with your Weaviate URL
    auth_client_secret=weaviate.AuthApiKey("YOUR-WEAVIATE-API-KEY"),  # If authentication is on. Replace w/ your Weaviate instance API key
)

# ==============================
# ===== BASIC GET EXAMPLES =====
# ==============================

# BasicGetPython
result = client.query.get("JeopardyQuestion", ["question"]).do()
print(result)
# END BasicGetPython

# Test results
assert "JeopardyQuestion" in result["data"]["Get"]
assert result["data"]["Get"]["JeopardyQuestion"][0].keys() == {"question"}
# End test


expected_result = """
// BasicGet Expected Results
{
  "data": {
    "Get": {
      "JeopardyQuestion": [
        {
          "question": "This prophet passed the time he spent inside a fish offering up prayers"
        },
        // shortened for brevity
      ]
    }
  }
}
// END BasicGet Expected Results
"""


gql_query = """
# BasicGetGraphQL
{
  Get {
    JeopardyQuestion {
      question
    }
  }
}
# END BasicGetGraphQL
"""
gqlresult = client.query.raw(gql_query)
# Test results
assert gqlresult == result
# END Test results



# ====================================
# ===== BASIC GET LIMIT EXAMPLES =====
# ====================================

# GetWithLimitPython
result = client.query.get("JeopardyQuestion", ["question"]).with_limit(1).do()
print(result)
# END GetWithLimitPython

# Test results
assert "JeopardyQuestion" in result["data"]["Get"]
assert len(result["data"]["Get"]["JeopardyQuestion"]) == 1
assert result["data"]["Get"]["JeopardyQuestion"][0].keys() == {"question"}
# End test


expected_result = """
// GetWithLimit Expected Results
{
  "data": {
    "Get": {
      "JeopardyQuestion": [
        {
          "question": "This prophet passed the time he spent inside a fish offering up prayers"
        },
        // Note this will only have one result as we limited it to 1
      ]
    }
  }
}
// END GetWithLimit Expected Results
"""


gql_query = """
# GetWithLimitGraphQL
{
  Get {
    JeopardyQuestion (limit: 1) {
      question
    }
  }
}
# ENDGetWithLimit GraphQL
"""
gqlresult = client.query.raw(gql_query)
# Test results
assert gqlresult == result
# END Test results



# ==========================================
# ===== GET LIMIT WITH OFFSET EXAMPLES =====
# ==========================================

# GetWithLimitOffsetPython
result = client.query.get("JeopardyQuestion", ["question"]).with_limit(1).with_offset(1).do()
print(result)
# END GetWithLimitOffsetPython

# Test results
assert "JeopardyQuestion" in result["data"]["Get"]
assert len(result["data"]["Get"]["JeopardyQuestion"]) == 1
assert result["data"]["Get"]["JeopardyQuestion"][0].keys() == {"question"}
# End test


expected_result = """
// GetWithLimitOffset Expected Results
{
  "data": {
    "Get": {
      "JeopardyQuestion": [
        {
          "question": "Pythons are oviparous, meaning they do this"
        }
      ]
    }
  }
}
// END GetWithLimitOffset Expected Results
"""


gql_query = """
# GetWithLimitOffsetGraphQL
{
  Get {
    JeopardyQuestion (limit: 1 offset: 1) {
      question
    }
  }
}
# END GetWithLimitOffsetGraphQL
"""
gqlresult = client.query.raw(gql_query)
# Test results
assert gqlresult == result
# END Test results



# ==========================================
# ===== GET OBJECT PROPERTIES EXAMPLES =====
# ==========================================

# GetPropertiesPython
result = client.query.get("JeopardyQuestion", ["question", "answer", "points"]).with_limit(1).do()
print(result)
# END GetPropertiesPython

# Test results
assert "JeopardyQuestion" in result["data"]["Get"]
assert len(result["data"]["Get"]["JeopardyQuestion"]) == 1
assert result["data"]["Get"]["JeopardyQuestion"][0].keys() == {"question", "answer", "points"}
# End test


expected_result = """
// GetProperties Expected Results
{
  "data": {
    "Get": {
      "JeopardyQuestion": [
        {
          "answer": "Jonah",
          "points": 100,
          "question": "This prophet passed the time he spent inside a fish offering up prayers"
        },
      ]
    }
  }
}
// END GetProperties Expected Results
"""


gql_query = """
# GetPropertiesGraphQL
{
  Get {
    JeopardyQuestion (limit: 1) {
      question
      answer
      points
    }
  }
}
# END GetPropertiesGraphQL
"""
gqlresult = client.query.raw(gql_query)
# Test results
assert gqlresult == result
# END Test results



# ======================================
# ===== GET OBJECT VECTOR EXAMPLES =====
# ======================================

# GetObjectVectorPython
result = client.query.get("JeopardyQuestion").with_additional("vector").with_limit(1).do()
print(result)
# END GetObjectVectorPython

# Test results
assert "JeopardyQuestion" in result["data"]["Get"]
assert len(result["data"]["Get"]["JeopardyQuestion"]) == 1
assert result["data"]["Get"]["JeopardyQuestion"][0]["_additional"].keys() == {"vector"}
# End test


expected_result = """
// GetObjectVector Expected Results
{
  "data": {
    "Get": {
      "JeopardyQuestion": [
        {
          "_additional": {
            "vector": [
              0.0065065133,
              -0.017786196,
              0.005879146,
              0.006707012,
              ...  // shortened for brevity
            ]
          }
        },
      ]
    }
  }
}
// END GetObjectVector Expected Results
"""


gql_query = """
# GetObjectVectorGraphQL
{
  Get {
    JeopardyQuestion (limit: 1) {
      _additional {
        vector
      }
    }
  }
}
# END GetObjectVectorGraphQL
"""
gqlresult = client.query.raw(gql_query)
# Test results
assert gqlresult == result
# END Test results


# ==================================
# ===== GET OBJECT ID EXAMPLES =====
# ==================================

# GetObjectIdPython
result = client.query.get("JeopardyQuestion").with_additional("id").with_limit(1).do()
print(result)
# END GetObjectIdPython

# Test results
assert "JeopardyQuestion" in result["data"]["Get"]
assert len(result["data"]["Get"]["JeopardyQuestion"]) == 1
assert result["data"]["Get"]["JeopardyQuestion"][0]["_additional"].keys() == {"id"}
# End test


expected_result = """
// GetObjectId Expected Results
{
  "data": {
    "Get": {
      "JeopardyQuestion": [
        {
          "_additional": {
            "id": "0002bf92-80c8-5d94-af34-0d6c5fea1aaf"
          }
        },
        // shortened for brevity
      ]
    }
  }
}
// END GetObjectId Expected Results
"""


gql_query = """
# GetObjectIdGraphQL
{
  Get {
    JeopardyQuestion (limit: 1) {
      _additional {
        id
      }
    }
  }
}
# END GetObjectIdGraphQL
"""
gqlresult = client.query.raw(gql_query)
# Test results
assert gqlresult == result
# END Test results
